lib/matplotlib/text.py

```diff
@@ -2072,4 +2082,13 @@ def get_tightbbox(self, renderer=None):
         return super().get_tightbbox(renderer)
 
 
+    def _check_xytext(self, renderer=None):
+        """Check whether the annotation at *xy_pixel* should be drawn."""
+        valid = True
+        if self.xytext is None:
+            return valid
+
+        if all(isinstance(xyt, numbers.Number) for xyt in self.xytext):
+            valid = not np.isnan(self.xytext).any() and np.isfinite(self.xytext).all()
```

Comment:
To fix the polar case, you need to check these values after they are transformed to "screen space", for which you can use `self.axes.transData.transform`.  See the [Transforms Tutorial](https://matplotlib.org/stable/users/explain/artists/transforms_tutorial.html) for more details.

I'm not really sure if invalid values here should make the annotation invisible or raise a more helpful error message.